File: WebshopBackend/WebshopBackend/Dtos/RegisterDtos.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebshopBackend.Dtos
{
    public class RegisterDtos
    {
        public long Id { get; set; }
        public string Username { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Email { get; set; }
        public string Password { get; set; }
        public string Role { get; set; }
        public string Street { get; set; }
        public string City { get; set; }
        public int Zip { get; set; }
        public string Country { get; set; }
    }
}

File: Frontend/Webshop/src/app/Dtos/RegisterDtos.ts
import {UserModel} from '../models/user.model'

export class RegisterDtos {
    
    Username : string;
    Password:string;
    FirstName: string;
    LastName: string;
    Role: string;
    Email: string;
    Street: string;
    City: string;
    Zip: number;
    Country: string

    constructor() {
    }

    CopyDataFromUser(user:UserModel)
    {
    this.Username=user.userName;
    this.Password=user.password;
    this.FirstName=user.firstName;
    this.LastName=user.lastName;
    this.Role="User";
    this.Email=user.email;
    this.Street=user.adress.street;
    this.City=user.adress.city;
    this.Country=user.adress.country;
    this.Zip=user.adress.zip;
    }
    
  }